### ISSUE TITLE
[BUG]  False positive when asserting an error pointer

### ISSUE THREAD

**Issue by nunnatsa** [nunnatsa/ginkgolinter#183]
**Describe the bug**
When asserting a pointer-error with `HaveOccurred` or  `Succeed`, ginkgolinter detect the value as non error, and triggers a linter error.

**To Reproduce**
```go
type myErr struct {
    code int
}

func (e *myErr) Error() string {
    ...
}

...
err = &myErr{code: 404}
Expect(err).To(HaveOccured())
```

**Expected behavior**
no linter error

**Actual behavior**
ginkgolinter mistakely detects `err` as not error type.

**Environment:**
 - OS: linux
 - Version v0.18.3

**Additional context**
Originally found here: https://github.com/containernetworking/cni/issues/1148

**Comment by nunnatsa** [comment]
Released version with the fix: v0.18.4

**Comment by nunnatsa** [comment]
[golangci-lint v1.63.0](https://github.com/golangci/golangci-lint/releases/tag/v1.63.0) was released with this fix
